src/Compilers/CSharp/Test/Syntax/Parsing/CollectionLiteralParsingTests.cs

```diff
@@ -0,0 +1,3223 @@
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "x");
+                        }
+                        N(SyntaxKind.QuestionQuestionToken);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "y");
+                        }
+                    }
+                }
+                N(SyntaxKind.CloseBracketToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestNullSuppression()
+        {
+            UsingExpression("[A]!");
+
+            N(SyntaxKind.SuppressNullableWarningExpression);
+            {
+                N(SyntaxKind.CollectionCreationExpression);
+                {
+                    N(SyntaxKind.OpenBracketToken);
+                    N(SyntaxKind.ExpressionElement);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "A");
+                        }
+                    }
+                    N(SyntaxKind.CloseBracketToken);
+                }
+                N(SyntaxKind.ExclamationToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestPreIncrement()
+        {
+            UsingExpression("++[A]");
+
+            N(SyntaxKind.PreIncrementExpression);
+            {
+                N(SyntaxKind.PlusPlusToken);
+                N(SyntaxKind.CollectionCreationExpression);
+                {
+                    N(SyntaxKind.OpenBracketToken);
+                    N(SyntaxKind.ExpressionElement);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "A");
+                        }
+                    }
+                    N(SyntaxKind.CloseBracketToken);
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestPostIncrement()
+        {
+            UsingExpression("[A]++");
+
+            N(SyntaxKind.PostIncrementExpression);
+            {
+                N(SyntaxKind.CollectionCreationExpression);
+                {
+                    N(SyntaxKind.OpenBracketToken);
+                    N(SyntaxKind.ExpressionElement);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "A");
+                        }
+                    }
+                    N(SyntaxKind.CloseBracketToken);
+                }
+                N(SyntaxKind.PlusPlusToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestAwaitParsedAsElementAccess()
+        {
+            UsingExpression("await [A]");
```

Comment:
Can you test this in top-level statements as well?

Reply:
done.